core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3401,11 +3409,17 @@ public PhysicalOperation visitTableExecute(TableExecuteNode node, LocalExecution
             return new PhysicalOperation(operatorFactory, outputMapping.buildOrThrow(), context, source);
         }
 
-        private int getWriterCount(Session session, Optional<PartitioningScheme> partitioningScheme, PlanNode source)
+        private int getWriterCount(Session session, Optional<PartitioningScheme> partitioningScheme, Optional<PartitioningScheme> preferredPartitioningScheme, PlanNode source)
         {
+            if (isLocalScaledWriterExchange(source)) {
+                return partitioningScheme.or(() -> preferredPartitioningScheme)
```

Comment:
nit: consider adding comment why we use `getTaskPartitionedWriterCount` when both `isLocalScaledWriterExchange` and `preferredPartitioningScheme` are set, because it might not be obvious for reader here